cli/command/engine/activate.go

```diff
@@ -0,0 +1,176 @@
+package engine
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/formatter"
+	"github.com/docker/cli/cli/containerizedengine"
+	"github.com/docker/cli/cli/licenseutils"
+	"github.com/docker/licensing/model"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type activateOptions struct {
+	licenseFile    string
+	version        string
+	registryPrefix string
+	format         string
+	image          string
+	quiet          bool
+}
+
+// NewActivateCommand creates a new `docker engine activate` command
+func newActivateCommand(dockerCli command.Cli) *cobra.Command {
+	var options activateOptions
+
+	cmd := &cobra.Command{
+		Use:   "activate [OPTIONS]",
+		Short: "Activate Enterprise Edition",
+		Long: `Activate Enterprise Edition.
+
+With this command you may apply an existing Docker enterprise license, or
+interactively download one from Docker.  In the interactive exchange, you can
+sign up for a new trial, or download an existing license.  If you are currently
+running a Community Edition engine, the daemon will be updated to the
+Enterprise Edition Docker engine with additional capabilities and long term
+support.
+
+For more information about different Docker Enterprise license types visit
+http://www.docker.com/licenses
+
+For non-interactive scriptable deployments, download your license from
+https://hub.docker.com/ then specify the file with the '--license' flag.
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runActivate(dockerCli, options)
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.StringVar(&options.licenseFile, "license", "", "License File")
+	flags.StringVar(&options.version, "version", "", "Specify engine version (default is to use existing version)")
+	flags.StringVar(&options.registryPrefix, "registry-prefix", "docker.io/docker", "Override the default location where engine images are pulled")
```

Comment:
Wouldn't this make more sense to have it as just `docker.io` and let the `engine-image` contain the prefix for the registry organization?

Reply:
This applies to all other uses of the `registry-prefix`

Reply:
I think this boils down to a question of use-case and when these flags will be used.  I was trying to optimize the UX around end-user use-cases to support local mirrors, while enabling developer use-cases.  

Specifically, if you're a typical user, I don't think you'll ever use `--engine-image`  - the default behavior for the community engine and enterprise engine scenarios will do the right thing.  This flag is really only meant for developers who want to deviate from the standard naming scheme.

Many customers run an internal registry in their firewalled environments, and the intention with the `--registry-prefix` flag was to give them a flag to establish the prefix where they store all their docker product mirrored images, both community and enterprise engines, and in the future, additional products like UCP and DTR itself.

My concern with splitting it the way you suggest is I think it makes it harder for end-users using local registry mirrors to benefit from the automatic community/enterprise logic.  These mirroring customer would always have to pin the exact image name to use a mirror.  In this version of the engine, we're using a single image so it seems ~simple, but it's entirely possible as we work on the modularization work in moby (buildkit, etc.) we will start to split the engine into multiple images in the future.  When/if we do so, then we'd have to add new flags for developers to select the other component images, but one could see this getting out of control for a typical user.  By using a common prefix and letting the CLI select the right image names by default using the common prefix we hide that complexity of multiple component images from the typical end user.  It also will make it harder for us to support this mirroring pattern for other products down the road. For example, UCP today doesn't support installing from a mirror, and UCP itself is made up of a dozen plus images already.  I'd like to see us follow this pattern so someday you can `docker ucp install --registry-prefix mylocalregistry.acme.com/docker ...` (or something along those lines) where the registry-prefix is common for all the mirrored content.